pkg/blockstorage/zone/zone.go

```diff
@@ -79,12 +78,14 @@ func getAvailableZones(ctx context.Context, kubeCli kubernetes.Interface, validZ
 	if availableRegion != sourceRegion {
 		log.Info().Print("Source region and available region mismatch", field.M{"sourceRegion": sourceRegion, "availableRegion": availableRegion})
 	}
-	if len(availableZones) <= 0 { // Will never occur, NodeZonesAndRegion returns error if empty
-		log.Info().Print("No available zones found", field.M{"availableRegion": availableRegion})
+	// TODO: validate availableRegion
+	validZoneNames, err := m.FromRegion(ctx, availableRegion)
+	if err != nil || len(validZoneNames) == 0 {
```

Comment:
what happens when `err != nil`?

Reply:
That is when we fail to get any zones from the available region (from k8s). We defer to using the source region.

Reply:
Makes sense. Please add a comment since we're dropping the error.